### DISCUSSION TITLE
Get triggers from Sybase

### DISCUSSION THREAD

**Discussion by JessicaF** [schemacrawler/SchemaCrawler#1430] [Q&A] [answered]
How do you retrieve triggers from a Sybase database? 

Currently I am attempting to do so with the following code which is an extension of the `DatabaseConnector` class:

```
informationSchemaViewsBuilder
      .withSql( 
        InformationSchemaKey.TRIGGERS,
              "SELECT\n" +
                  "    db_name() AS TRIGGER_CATALOG,\n" +
                  "    user_name(o.uid) as TRIGGER_SCHEMA,\n" +
                  "    o.name as TRIGGER_NAME,\n" +
                  "    NULL as SPECIFIC_NAME,\n" +
                  "    c.text as ACTION_STATEMENT\n" +
                  "FROM syscomments c\n" +
                  "         JOIN sysobjects o\n" +
                  "              ON c.id = o.id\n" +
                  "WHERE o.type ='TR'\n" +
                  "ORDER BY o.name"
                  )
```

It currently works with routines and views but not with triggers.

**Comment by sualeh** [comment, marked as answer]
@JessicaF If you are doing extensive customization like this, it is a good idea to write your own SchemaCrawler database plugin instead of wrangling the API like this. You can use the (unsupported) [SchemaCrawler plugin for Sybase](https://github.com/schemacrawler/SchemaCrawler-Database-Plugins/tree/main/schemacrawler-sapiq) as your starting point. In fact, I would request you to commit to that project directly, so others can benefit too.
